src/actions/party.ts

```diff
@@ -1,73 +1,116 @@
+                    return party;
+                })
+                .then(party => dispatch(this.setPartyRecieved(party)))
+                .catch(e => console.log(e));
+        };
+    }
 
-export function getPartyInfo() {
-    return dispatch => {
-        dispatch(NavigationActions.navigate({routeName: 'JoinParty'}));
+    getInfo() {
+        return dispatch => {
+            dispatch(NavigationActions.navigate({ routeName: 'JoinParty' }));
+        }
     }
-}
 
-export function setPartyRecieved(party) {
-    return {
-        type: types.PARTY_RECIEVED,
-        party: party
-    };
-}
+    kickPlayer(id) {
+        return getState => {
+            const { user, party } = getState();
 
-export function subscribeToParty({id}) {
-    return (dispatch) => {
-        Party.subscribe(
-            id,
-            user => dispatch(setUserJoined(user)),
-            userId => dispatch(setUserFled(userId))
-        );
-    };
-}
+            Party.kickPlayer({
+                curentPartyId: party.id,
+                kickedId: id,
+                currentUserId: user.id
+            });
+        }
+    }
+
+    promotePlayer(id) {
+        return getState => {
+            const { user, party } = getState();
+
+            Party.promotePlayer({
+                curentPartyId: party.id,
+                kickedId: id,
+                currentUserId: user.id
+            });
+        }
+    }
+
+    private setPartyRecieved(party) {
+        return {
+            type: ActionTypes.PARTY_RECIEVED,
+            party: party
+        };
+    }
+
+    private subscribeToParty({ id }) {
+        return (dispatch) => {
+            Party.subscribe(
+                id,
+                user => dispatch(this.setUserJoined(user)),
+                userId => dispatch(this.setUserFled(userId)),
+                userId => dispatch(this.setUserKicked(userId)),
+                userId => dispatch(this.setUserPromote(userId))
+            );
+        };
+    }
+
+    private setUserJoined(joinedUser) {
+        return (dispatch, getState) => {
+            const { user } = getState();
+            if (joinedUser.id === user.id)
+                return;
+            dispatch({
+                type: ActionTypes.USER_JOINED,
+                user: joinedUser
+            });
+        }
+    }
+
+    private setUserFled({ id }) {
+        return {
+            type: ActionTypes.USER_FLED,
+            id
+        };
+    }
+
+    private setUserKicked({ id }) {
+        return {
+            type: ActionTypes.USER_KICKED,
+            id
+        };
+    }
 
-export function setUserJoined(joinedUser) {
-    return (dispatch, getState) => {
-        const { user } = getState();
-        if(joinedUser.id === user.id)
-            return;
-        dispatch({
-            type: types.USER_JOINED,
-            user: joinedUser
-        });
+    private setUserPromote({ id }) {
+        return {
+            type: ActionTypes.USER_PROMOTE,
+            id
+        };
```

Comment:
this sounds like a lot of boilerplate, I wonder if we can refactor this to be easier, and we don't have to create a method for each action. the only thing that changes in these objects is the type, so we can probably have a generic method